internal/rbd/rbd_util.go

```diff
@@ -603,36 +603,31 @@ func (rv *rbdVolume) getCloneDepth(ctx context.Context) (uint, error) {
 	vol.Monitors = rv.Monitors
 	vol.RbdImageName = rv.RbdImageName
 	vol.conn = rv.conn.Copy()
-
+	currVol := &vol
```

Comment:
use `vol := &rbdVolume{}` instead?

There needs to be a `defer vol.Destroy()` somewhere (or `currVol.Destroy()`, but `vol.Destroy()` will be less confusing).